lib/connectors_shared/middleware/restrict_hostnames.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+require 'faraday/middleware'
+
```

Comment:
I bet this is missing a lot of `require` statements to find classes like `AddressNotAllowed`, `Resolv`, `IPAddr`, etc